include/SDL3/SDL_audio.h

```diff
@@ -292,6 +295,8 @@ typedef struct SDL_AudioSpec
     SDL_AudioFormat format;     /**< Audio data format */
     int channels;               /**< Number of channels: 1 mono, 2 stereo, etc */
     int freq;                   /**< sample rate: sample frames per second */
+    SDL_bool use_channel_map;   /**< If SDL_FALSE, ignore `channel_map` and use default order. */
+    Uint8 channel_map[16];      /**< `channels` items of channel order. */
```

Comment:
So this provides a channel map from the audio data to SDL's expected channel layout, but does it actually make sense to have an array of channels in the order they're expected? That would be a complete description of the audio data rather than implying knowledge of the default channel map (which we might want to change someday?)

Here's how Microsoft represents the channel map:
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ksmedia/ns-ksmedia-ksaudio_channel_config
For them, the channels are always in a specific order, and a bitmask tells which channels are present.

Reply:
Yeah, you're right. Maybe we also need a way to know what channels are being provided? e.g. this is 7.1 home theater with channels "FL, FR, FC, LFE, BL, BR, SL, and SR", or this is 7.1 wide configuration with channels "FL, FR, FC, LFE, BL, BR, FLC, and FRC"?

Reply:
In any case, I think what we have here is good to go.